core/services/relay/evm/mercury/v1/data_source.go

```diff
@@ -290,40 +293,36 @@ func (ds *datasource) executeRun(ctx context.Context) (*pipeline.Run, pipeline.T
 	return run, trrs, err
 }
 
-func (ds *datasource) setLatestBlocks(ctx context.Context, obs *relaymercuryv1.Observation) {
-	latestBlocks := ds.getLatestBlocks(ctx, nBlocksObservation)
+func (ds *datasource) setLatestBlocks(ctx context.Context, obs *relaymercuryv1.Observation) error {
+	latestBlocks, err := ds.chainReader.LatestHeads(ctx, nBlocksObservation)
+	if err != nil {
+		ds.lggr.Errorw("failed to read latest blocks", "error", err)
```

Comment:
Probably this log line is redundant since the Observe error will be logged further up the stack but not a big deal since its so rare